src/ashell/comms-uart.c

```diff
@@ -322,31 +303,18 @@ void comms_write_buf(const char *buf, int len)
     if (len == 0)
         return;
 
-    struct device *dev = dev_upload;
-    uart_irq_tx_enable(dev);
-
+    uart_irq_tx_enable(dev_upload);
     data_transmitted = false;
-    uart_fifo_fill(dev, buf, len);
-
+    uart_fifo_fill(dev_upload, (const uint8_t *)buf, len);
     while (data_transmitted == false);
-
-    uart_irq_tx_disable(dev);
-}
-
-void comms_writec(char byte)
-{
-    comms_write_buf(&byte, 1);
-}
-
-void comms_print(const char *buf)
-{
-    comms_write_buf(buf, strnlen(buf, MAX_LINE_LEN * 4));
+    uart_irq_tx_disable(dev_upload);
 }
-
-void comms_println(const char *buf)
+void comms_print(const char *buf, bool newline)
```

Comment:
Agreed, that was leftover from testing.  And yes I don't want to have several comms_write_* functions, it makes it confusing (to me at least) what exactly is being done in each.